local-server/local-server-services/src/main/java/com/esri/samples/localserver/LocalServerServicesController.java

```diff
@@ -0,0 +1,218 @@
+              }
+            });
+            mapService.startAsync();
+            break;
+          case "Feature Service":
+            String featureServiceURL = Paths.get(getClass().getResource("/PointsofInterest.mpk").toURI()).toString();
+            LocalFeatureService featureService = new LocalFeatureService(featureServiceURL);
+            featureService.addStatusChangedListener(status -> {
+              centerText.appendText("Feature Service Status: " + status.getNewStatus().toString() + "\n");
+              if (status.getNewStatus() == LocalServerStatus.STARTED) {
+                //              bottomText.appendText("Map Service " + mapService.getUrl());
+                Platform
+                    .runLater(() -> bottomView.getItems().add("Feature Service URL ->  " + featureService.getUrl()));
+              }
+            });
+            featureService.startAsync();
+            break;
+          case "Geoprocessing Service":
+            String geoServiceURL = Paths.get(getClass().getResource("/MessageInABottle.gpk").toURI()).toString();
+            LocalGeoprocessingService geoService = new LocalGeoprocessingService(geoServiceURL);
+            geoService.addStatusChangedListener(status -> {
+              centerText.appendText("Geo Service Status: " + status.getNewStatus().toString() + "\n");
+              if (status.getNewStatus() == LocalServerStatus.STARTED) {
+                //              bottomText.appendText("Map Service " + mapService.getUrl());
+                Platform.runLater(() -> bottomView.getItems().add("Geo Service URL ->  " + geoService.getUrl()));
+              }
+            });
+            geoService.startAsync();
+            break;
+        }
+      } else {
+        Platform.runLater(() -> {
+          Alert dialog = new Alert(AlertType.INFORMATION);
+          dialog.setHeaderText("Service Running");
+          dialog.setContentText(service + " has already been started.");
+          dialog.showAndWait();
+        });
+      }
+    }
+  }
+
+  @FXML
+  protected void handleStopService(ActionEvent e) {
+    int serviceIndex = bottomView.getSelectionModel().getSelectedIndex();
+    services.get(serviceIndex).stopAsync();
+    bottomView.getItems().remove(serviceIndex);
+
+  }
+
+  @FXML
+  protected void enableButtons() {
+
+    int listIndex = bottomView.getSelectionModel().getSelectedIndex();
+    if (listIndex != -1) {
+      btnStopService.setDisable(false);
+      btnURL.setDisable(false);
+    } else {
+      btnStopService.setDisable(true);
+      btnURL.setDisable(true);
+    }
+  }
+
+  @FXML
+  protected void handleURL(ActionEvent e) {
+    if (services != null && services.size() > 0) {
+      String url = bottomView.getSelectionModel().getSelectedItem().split(">")[1].trim();
+      System.out.println("URL: " + url);
+      try {
+        Desktop.getDesktop().browse(new URI(url));

```

Comment:
This isn't the Java FX way - you need to use https://docs.oracle.com/javase/8/javafx/api/javafx/application/HostServices.html